benches/shredder_benchmark.rs

```diff
@@ -0,0 +1,121 @@
+use std::cell::RefCell;
+
+use criterion::{black_box, Criterion, criterion_main};
+use criterion::criterion_group;
+use rand::rngs::StdRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use shredder::{collect, number_of_tracked_allocations, Gc, Scan, run_with_gc_cleanup};
+
+// BENCHMARK 1: My janky stress test
+#[derive(Debug, Scan)]
+struct DirectedGraphNode {
+    label: String,
+    edges: Vec<Gc<RefCell<DirectedGraphNode>>>,
+}
+
+const NODE_COUNT: usize = 1 << 15;
+const EDGE_COUNT: usize = 1 << 15;
+const SHRINK_DIV: usize = 1 << 10;
+
+fn stress_test() -> Vec<usize> {
```

Comment:
Am I correct in my reading that this is just creating a bunch of nodes, then edges between those nodes, and then removing them?

Reply:
Yes